src/Compilers/Core/MSBuildTask/Microsoft.CSharp.Core.targets

```diff
@@ -3,10 +3,21 @@
 <Project ToolsVersion="14.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <Import Project="Microsoft.Managed.Core.targets"/>
 
-  <PropertyGroup Condition="('$(TargetFrameworkIdentifier)' != '.NETCoreApp' OR '$(_TargetFrameworkVersionWithoutV)' &lt; '3.0') AND
-                            ('$(TargetFrameworkIdentifier)' != '.NETStandard' OR '$(_TargetFrameworkVersionWithoutV)' &lt; '2.1')">
-    <MaxSupportedLangVersion Condition="'$(MaxSupportedLangVersion)' == ''">7.3</MaxSupportedLangVersion>
-    <LangVersion Condition="'$(LangVersion)' == ''">$(MaxSupportedLangVersion)</LangVersion>
+  <PropertyGroup>
+    <!-- .NETCoreApp < 3.0, .NETStandard < 2.1, or any other target framework -->
+    <_MaxSupportedLangVersion Condition="('$(TargetFrameworkIdentifier)' != '.NETCoreApp' OR '$(_TargetFrameworkVersionWithoutV)' &lt; '3.0') AND
```

Comment:
>_TargetFrameworkVersionWithoutV [](start = 96, length = 31)

`_TargetFrameworkVersionWithoutV` is a private property defined in .NET SDK. We shouldn't use private properties defined outside of Roslyn targets.

Reply:
The check implemented here is the one the SDK team recommended that we use. If there is a better property to use here we'd be happy to do so. 

Note: @chsienki didn't add the use of this property here, his change just reorganized the existing checks and built off of the pattern.

Reply:
We can copy the definition of this property to our targets:

```
<_TargetFrameworkVersionWithoutV>$(TargetFrameworkVersion.TrimStart('vV'))</_TargetFrameworkVersionWithoutV>
```

Reply:
MSBuild is improving this. I've filed https://github.com/dotnet/roslyn/issues/46050 to track